### PR TITLE
Add extension ChatClient::loadMessageById

### PR DESCRIPTION
**Pull request by amitkma**
https://github.com/GetStream/stream-chat-android/issues/2895

### 🎯 Goal

To merge `ChatDomain` into LLC, we need to have an extension over `ChatClient` that does hide the `ChatDomain` logic.

### 🛠 Implementation details

- A new extension function `ChatClient::loadMessageById` is created.
- All calls of `ChatDomain::loadMessageById` are replaced by new `ChatClient::loadMessageById` extension.
- Logic has been moved inside OfflinePlugin.

### 🧪 Testing

Run the UI and compose sample app and both should work as before with no behavior change.

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-team or #compose-chat-sdk-team) (required)
- [x] PR targets the `develop` branch

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt [hunk 1]
```diff
@@ -914,7 +914,13 @@ internal class ChatDomainImpl internal constructor(
         messageId: String,
         olderMessagesOffset: Int,
         newerMessagesOffset: Int,
-    ): Call<Message> = LoadMessageById(this).invoke(cid, messageId, olderMessagesOffset, newerMessagesOffset)
+    ): Call<Message> {
+        validateCid(cid)
+        val channelController = channel(cid)
```
Comment (JcMinarro) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt:919]: As far as we are returning a `Call<Message>` that represent a deferred computation, all this code should be performed inside of the call when it is run

Comment (amitkma) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt:919]: Yep, I agree.

Comment (amitkma) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/ChatDomainImpl.kt]: @JcMinarro Please check the changes.

**Review by MarinTolic — CHANGES_REQUESTED** [review]
Requested changes solely for the shadowing variable names, the comment suggestions are optional and up to you whether you want to implement them or not. :)

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt [hunk 2]
```diff
@@ -148,6 +150,64 @@ internal class ChannelLogic(
...
+        messageId: String,
+        olderMessagesOffset: Int,
+        newerMessagesOffset: Int,
+    ): Result<Message> {
+        return chatDomainImpl.repos.selectMessage(messageId)?.let { message ->
+            Result(message)
+        } ?: Result(ChatError("Error while fetching message from backend. Message id: $messageId"))
+    }
+
+    /** Loads a list of messages after the newest message in the current list. */
```
[the remark this answers is not in the document]
Comment (amitkma) [reply stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt:180]: Not sure, I picked this part of the code from our `ChannelController`.

**Review by MarinTolic — APPROVED** [review]
If there are attachments around the pinned message they can bump it up out of bounds and make it appear as if it didn't work. However this seems to be a UI issue related to the `RecyclerView` itself and not to the core parts.

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] ~~UI Changes correct (before & after images)~~
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Review by adasiewiczr — APPROVED** [review]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**